src/osd/PrimaryLogPG.cc

```diff
@@ -4458,8 +4532,44 @@ int PrimaryLogPG::trim_object(
     ctx->delta_stats.num_object_clones--;
     if (coi.is_cache_pinned())
       ctx->delta_stats.num_objects_pinned--;
-    if (coi.has_manifest())
+    if (coi.has_manifest()) {
+      // adjacent clones
```

Comment:
Isn't this the clone equivalent to dec_all_refcount_head_manifest()?  I'd like you to pull this out into a helper function.  I'm not actually sure why you can't unify it with dec_all_refcount_head_manifest().